x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/ExecuteStepsUpdateTask.java

```diff
@@ -102,8 +101,14 @@ public ClusterState execute(final ClusterState currentState) throws IOException
                     } else {
                         state = IndexLifecycleRunner.moveClusterStateToNextStep(index, state, currentStep.getKey(),
                             currentStep.getNextStepKey(), nowSupplier, false);
+                        // set here to make sure that the clusterProcessed knows to execute the
+                        // correct step if it an async action
+                        nextStepKey = currentStep.getNextStepKey();
```

Comment:
Can you move this to line 99 (before the `null` check) and then use the var so that we don't call `getNextStepKey()` three times, I want to avoid any situation in the future where subsequent calls could garner a different result.

Reply:
yes